ciao-cli/configuration.go

```diff
@@ -0,0 +1,237 @@
+	"github.com/01org/ciao/ciao-controller/types"
+)
+
+var configCommand = &command{
+	SubCommands: map[string]subCommand{
+		"update": new(configUpdateCommand),
+		"show":   new(configShowCommand),
+	},
+}
+
+type confElement string
+
+type configUpdateCommand struct {
+	Flag    flag.FlagSet
+	element confElement
+	value   string
+}
+
+func (e *confElement) String() string {
+	return string(*e)
+}
+
+func (e *confElement) Set(s string) error {
+	switch string(s) {
+	case "scheduler.storage_uri",
+		"controller.compute_port",
+		"launcher.disk_limit",
+		"launcher.mem_limit",
+		"identity_service.type",
+		"identity_service.url":
+
+		*e = confElement(s)
+		return nil
+	}
+	return fmt.Errorf("invalid element of the cluster configuration")
+
+}
+
+func (cmd *configUpdateCommand) usage(...string) {
+	fmt.Fprintf(os.Stderr, `
+Usage:
+
+ciao-cli [options] config update [flags]
+
+update cluster configuration
+
+The update flags are:
+
+`)
+	cmd.Flag.PrintDefaults()
+	os.Exit(2)
+}
+
+func (cmd *configUpdateCommand) parseArgs(args []string) []string {
+	cmd.Flag.Usage = func() { cmd.usage() }
+	cmd.Flag.Var(&cmd.element, "element", "configuration field to change")
+	cmd.Flag.StringVar(&cmd.value, "value", "", "new value for element to change")
+	cmd.Flag.Parse(args)
+	return cmd.Flag.Args()
+}
+
+func (cmd *configUpdateCommand) run(args []string) error {
+	var req types.ConfigRequest
+	var response types.ConfigUpdateResponse
+	if cmd.element == "" {
+		fmt.Fprintf(os.Stderr, "Missing required -element parameter\n")
+		cmd.usage()
+	}
+
+	if cmd.value == "" {
+		fmt.Fprintf(os.Stderr, "Missing required -value parameter\n")
+		cmd.usage()
+	}
+	if validConfigValue(cmd.value, string(cmd.element)) == false {
+		return fmt.Errorf("'%s' is not a valid value for %s", cmd.value, cmd.element)
+	}
+
+	url := buildCiaoURL("configuration")
+	req = types.ConfigRequest{
+		Element: string(cmd.element),
+		Value:   cmd.value,
+	}
+	b, err := json.Marshal(req)
+	if err != nil {
+		fatalf(err.Error())
+	}
+	body := bytes.NewReader(b)
+	ver := api.ClusterConfigV1
+
+	resp, err := sendCiaoRequest("POST", url, nil, body, &ver)
+	if err != nil {
+		return err
+	}
+
+	err = unmarshalHTTPResponse(resp, &response)
+	if err != nil {
+		return err
+	}
+	fmt.Println(response.Response)
+	return nil
+}
+
+func validConfigValue(s string, t string) bool {
+	switch t {
+	case "scheduler.storage_uri":
+		return validConfigURI(s, "file")
+	case "identity_service.url":
+		return validConfigURI(s, "https")
+	case "controller.compute_port":
+		return validConfigNumber(s)
+	case "launcher.disk_limit", "launcher.mem_limit":
+		return validConfigBoolean(s)
+	case "identity_service.type":
+		return s == "keystone"
+	}
+	return false
+}
+
+// validConfigNumber checks that current input is a sane integer number
+func validConfigNumber(s string) bool {
+	if _, err := strconv.Atoi(s); err == nil {
+		return true
+	}
+	return false
+}
+
+func sanatizeBoolean(s string) string {
```

Comment:
Sorry, my mistake here, the idea is to sanitize those values previous to send the request, so this function will be called when creating it.